m4/odp_check_flag.m4

```diff
@@ -1,7 +1,5 @@
-# Copyright (c) 2018, Linaro Limited
-# All rights reserved.
-#
-# SPDX-License-Identifier:     BSD-3-Clause
+# SPDX-License-Identifier: BSD-3-Clause
```

Comment:
There is a bunch of `m4` files that don't have a copyright listed at all, I guess we should tag them as well?

Reply:
Added copyrights in V2.

Reply:
`/platform/linux-generic/m4/` hides a few additional `m4`s.

Reply:
Yeah, I was on the fence about these, since many of them are pretty trivial, but it makes sense to have license text on all files. Added in V3.